### PR TITLE
z_play retail OK

### PR DESCRIPTION
**Pull request by engineer124**
Also updates the `SET_NEXT_GAMESTATE` to have a local temp and fixes all the function that broke because of it.

### CODE REVIEW

File: src/code/z_play.c [hunk 1]
```diff
@@ -541,11 +561,15 @@ void Play_Update(PlayState* this) {
...
 
+#if OOT_DEBUG
                     if (!R_TRANS_DBG_ENABLED) {
                         Play_SetupTransition(this, this->transitionType);
                     } else {
                         Play_SetupTransition(this, R_TRANS_DBG_TYPE);
                     }
+#else
+                    Play_SetupTransition(this, this->transitionType);
+#endif
```
Comment (engineer124) [inline src/code/z_play.c:572]: This particular block was picky and didn't want to accept `if (!OOT_DEBUG || !R_TRANS_DBG_ENABLED) {`, a better match is welcome

File: src/code/z_play.c [hunk 2]
```diff
@@ -18,12 +18,16 @@ void Play_SpawnScene(PlayState* this, s32 sceneId, s32 spawn);
...
+#if OOT_DEBUG
 #define PLAY_LOG(line)                            \
     do {                                          \
         if (R_ENABLE_PLAY_LOGS) {                 \
             LOG_NUM("1", 1, "../z_play.c", line); \
         }                                         \
     } while (0)
+#else
+#define PLAY_LOG(line) (void)0
+#endif
```
Comment (fig02) [inline src/code/z_play.c:30]: does the debug check in the if work?

File: src/code/z_play.c (hunk 1 above)
Comment (fig02) [reply src/code/z_play.c:572]: does it match with the debug check second, we've seen a couple cases of that already

File: src/code/z_play.c [hunk 3]
```diff
@@ -156,7 +160,11 @@ void Play_SetupTransition(PlayState* this, s32 transitionType) {
                 break;
 
             default:
+#if OOT_DEBUG
                 Fault_AddHungupAndCrash("../z_play.c", 2290);
+#else
+                LogUtils_HungupThread("../z_play.c", 2290);
+#endif
```
Comment (cadmic) [inline src/code/z_play.c:167]:
```suggestion
                HUNGUP_AND_CRASH("../z_play.c", 2290);
```

File: src/code/z_play.c (hunk 1 above)
Comment (engineer124) [reply src/code/z_play.c:568]: It doesn't unfortunately. I've only seen the debug check second work for `&& OOT_DEBUG`, has it also worked for `|| !OOT_DEBUG`?

File: src/code/z_play.c (hunk 2 above)
Comment (engineer124) [reply src/code/z_play.c:30]: Not at first glance, not even as `&& OOT_DEBUG`. Breaks matching in a few place, both retail and debug

**Review by fig02 — APPROVED** [review]

File: src/code/z_play.c (hunk 1 above)
Comment (cadmic) [reply src/code/z_play.c:568]: There's an `|| !OOT_DEBUG` that's like that in game.c currently (there's two, but one of them matches the other way too, and I don't remember which)

**Review by cadmic — APPROVED** [review]

**Pull request merged**